src/khoj/routers/helpers.py

```diff
@@ -175,35 +195,41 @@ async def generate_better_image_prompt(q: str, conversation_history: str) -> str
 
 async def send_message_to_model_wrapper(
     message: str,
+    system_message: str = "",
 ):
-    conversation_config = await ConversationAdapters.aget_default_conversation_config()
+    conversation_config: ChatModelOptions = await ConversationAdapters.aget_default_conversation_config()
 
     if conversation_config is None:
         raise HTTPException(status_code=500, detail="Contact the server administrator to set a default chat model.")
 
+    truncated_messages = generate_chatml_messages_with_context(
+        user_message=message, system_message=system_message, model_name=conversation_config.chat_model
+    )
+
     if conversation_config.model_type == "offline":
         if state.gpt4all_processor_config is None or state.gpt4all_processor_config.loaded_model is None:
             state.gpt4all_processor_config = GPT4AllProcessorModel(conversation_config.chat_model)
 
         loaded_model = state.gpt4all_processor_config.loaded_model
         return send_message_to_model_offline(
-            message=message,
+            message=truncated_messages[-1].content,
             loaded_model=loaded_model,
             model=conversation_config.chat_model,
             streaming=False,
+            system_message=truncated_messages[0].content,
```

Comment:
Is system message always the first of the `truncated_messages`? Why not just set system_message to system_message?

```suggestion
            system_message=system_message,
```

Reply:
We would probably want to leave an option for the truncation logic to act as an underlying abstraction layer to handle longer messages. I wouldn't want to override its decision if/when we apply some truncation logic to the system prompt.